tests/fixture/bootstrapmonitor/common.go

```diff
@@ -0,0 +1,294 @@
+// Copyright (C) 2019-2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package bootstrapmonitor
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"go.uber.org/zap"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/clientcmd"
+
+	"github.com/ava-labs/avalanchego/utils/logging"
+	"github.com/ava-labs/avalanchego/version"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	restclient "k8s.io/client-go/rest"
+)
+
+// Path to write the details to on the data volume
+func getTestDetailsPath(dataDir string) string {
+	return filepath.Join(dataDir, "bootstrap_test_details.txt")
+}
+
+// Used to serialize test details to the data volume used for a given test to
+// support resuming a previously started test and tracking test duration.
+type bootstrapTestDetails struct {
+	Image     string    `json:"image"`
+	StartTime time.Time `json:"startTime"`
+}
+
+// WaitForPodCondition watches the specified pod until the status includes the specified condition.
+func WaitForPodCondition(ctx context.Context, clientset *kubernetes.Clientset, namespace string, podName string, conditionType corev1.PodConditionType) error {
+	return waitForPodStatus(
+		ctx,
+		clientset,
+		namespace,
+		podName,
+		func(status *corev1.PodStatus) bool {
+			for _, condition := range status.Conditions {
+				if condition.Type == conditionType && condition.Status == corev1.ConditionTrue {
+					return true
+				}
+			}
+			return false
+		},
+	)
+}
+
+// waitForPodStatus watches the specified pod until the status is deemed acceptable by the provided test function.
+func waitForPodStatus(
+	ctx context.Context,
+	clientset *kubernetes.Clientset,
+	namespace string,
+	name string,
+	acceptable func(*corev1.PodStatus) bool,
+) error {
+	watch, err := clientset.CoreV1().Pods(namespace).Watch(ctx, metav1.SingleObject(metav1.ObjectMeta{Name: name}))
+	if err != nil {
+		return fmt.Errorf("failed to initiate watch of pod %s/%s: %w", namespace, name, err)
+	}
+
+	for {
+		select {
+		case event := <-watch.ResultChan():
+			pod, ok := event.Object.(*corev1.Pod)
+			if !ok {
+				continue
+			}
+
+			if acceptable(&pod.Status) {
+				return nil
+			}
+		case <-ctx.Done():
+			return errors.New("timeout waiting for pod readiness")
```

Comment:
not sure if this would be meaningful, but generally speaking, when you have an expired context, you want to return an error that is derived from context.ContextCanceled or similar.